src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -625,13 +635,25 @@ public TResult TryGetLast(out bool found)
                 return sourceFound ? _selector(input) : default(TResult);
             }
 
+            private TResult[] LazyToArray()
```

Comment:
@stephentoub, I separated the `default:` case into a new `PreallocatingToArray` method.